### ISSUE TITLE
VMAlert - Broken URLs for "System" endpoints

### ISSUE THREAD

**Issue by lukeod** [VictoriaMetrics/VictoriaMetrics#6603]
### Describe the bug

When I navigate to the web admin page for VMAlert, the "System" urls don't work. These are the "flags", "metrics", "-/reload" endpoints. I think this was broken in a recent update,  perhaps https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6435.

It seems that by default, VMAlert listens on :8880 and will serve the web console on both :8880/ and :8880/vmalert/.

The "System" links only work against the :8880/ root such as:

http://mydockerhost:8880/flags
http://mydockerhost:8880/metrics
http://mydockerhost:8880/-/reload

Whereas these don't work:

http://mydockerhost:8880/vmalert/flags
http://mydockerhost:8880/vmalert/metrics
http://mydockerhost:8880/vmalert/-/reload

Which throw errors like:
"remoteAddr: "10.10.10.10:62013"; requestURI: /vmalert/flags; unsupported path requested: "/vmalert/flags"

While the web console listens to both / and /vmalert, it seems to "prefer" /vmalert for most endpoints. For instance if you click the various links up the top (vmalert, groups, alerts) it uses the /vmalert path. This means that if you are navigating around the page, you inevitably end up back at the /vmalert/ page, where the admin links no longer work.

### To Reproduce

I replicated the issue with a simple docker-compose setup, which is based on the reference docker-compose.yml in the git repo:
```
services:
  victoriametrics:
    container_name: victoriametrics
    image: victoriametrics/victoria-metrics:stable
    ports:
      - 8428:8428
      - 8089:8089

      - 8089:8089/udp
      - 2003:2003
      - 2003:2003/udp
      - 4242:4242
    volumes:
      - vmdata:/storage
    command:
      - "--storageDataPath=/storage"
      - "--graphiteListenAddr=:2003"
      - "--opentsdbListenAddr=:4242"
      - "--httpListenAddr=:8428"
      - "--influxListenAddr=:8089"
      - "--vmalert.proxyURL=http://vmalert:8880"
    networks:
      - vm_net
    restart: always

  vmalert:
    container_name: vmalert
    image: victoriametrics/vmalert:stable
    depends_on:
      - "victoriametrics"
    ports:
      - 8880:8880
    volumes:
      - ./alerts/:/etc/alerts/
    command:
      - "--datasource.url=http://victoriametrics:8428/"
      - "--remoteRead.url=http://victoriametrics:8428/"
      - "--remoteWrite.url=http://victoriametrics:8428/"
      - "--notifier.url=http://alertmanager:9093/"
      - "--rule=/etc/alerts/*.yml"
      - '--external.alert.source=explore?orgId=1&left={"datasource":"VictoriaMetrics","queries":[{"expr":{{.Expr|jsonEscape|queryEscape}},"refId":"A"}],"range":{"from":"{{ .ActiveAt.UnixMilli }}","to":"now"}}'
    networks:
      - vm_net
    restart: always

volumes:
  vmdata: {}
networks:
  vm_net:
```
Navigate to http://mydockerhost:8880/, try and navigate to the "System" urls. They should work.

Now click on other links at the top such as "Groups", and then click the "vmalert" link in the upper left corner, which acts like a "home" link.

Now the url will be http://mydockerhost:8880/vmalert, and the system urls will no longer work. They now link to http://mydockerhost:8880/vmalert/flags, http://mydockerhost:8880/vmalert/metrics etc, which throw an error when you navigate to them.

### Version
```
/ # /vmalert-prod --version
vmalert-20240624-230104-tags-v1.102.0-rc2-0-gde7450b7e0
```
### Logs

```
root@dockerhost:/opt/victoriametrics# docker logs vmalert
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: vmalert-20240624-230104-tags-v1.102.0-rc2-0-gde7450b7e0
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:13   command-line flags
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -datasource.url="secret"
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -external.alert.source="explore?orgId=1&left={\"datasource\":\"VictoriaMetrics\",\"queries\":[{\"expr\":{{.Expr|jsonEscape|queryEscape}},\"refId\":\"A\"}],\"range\":{\"from\":\"{{ .ActiveAt.UnixMilli }}\",\"to\":\"now\"}}"
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -notifier.url="secret"
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -remoteRead.url="secret"
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -remoteWrite.url="secret"
2024-07-08T03:59:06.688Z        info    VictoriaMetrics/lib/logger/flag.go:20     -rule="/etc/alerts/*.yml"
2024-07-08T03:59:06.689Z        info    VictoriaMetrics/app/vmalert/main.go:165 reading rules configuration file from "/etc/alerts/*.yml"
2024-07-08T03:59:06.689Z        info    VictoriaMetrics/app/vmalert/config/log/logger.go:52     found 0 files to read from "Local FS{MatchPattern: \"/etc/alerts/*.yml\"}"
2024-07-08T03:59:06.689Z        warn    VictoriaMetrics/app/vmalert/config/log/logger.go:44     no groups found in /etc/alerts/*.yml
2024-07-08T03:59:06.689Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:120        starting server at http://127.0.0.1:8880/
2024-07-08T03:59:06.689Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:121        pprof handlers are exposed at http://127.0.0.1:8880/debug/pprof/
2024-07-08T03:59:34.755Z        info    VictoriaMetrics/app/vmalert/web.go:173  api config reload was called, sending sighup
2024-07-08T03:59:34.755Z        info    VictoriaMetrics/app/vmalert/main.go:355 SIGHUP received. Going to reload rules "[/etc/alerts/*.yml]" ...
2024-07-08T03:59:34.755Z        info    VictoriaMetrics/app/vmalert/config/log/logger.go:52     found 0 files to read from "Local FS{MatchPattern: \"/etc/alerts/*.yml\"}"
2024-07-08T03:59:34.756Z        warn    VictoriaMetrics/app/vmalert/config/log/logger.go:44     no groups found in /etc/alerts/*.yml
2024-07-08T04:06:08.720Z        info    VictoriaMetrics/app/vmalert/web.go:173  api config reload was called, sending sighup
2024-07-08T04:06:08.720Z        info    VictoriaMetrics/app/vmalert/main.go:355 SIGHUP received. Going to reload rules "[/etc/alerts/*.yml]" ...
2024-07-08T04:06:08.721Z        info    VictoriaMetrics/app/vmalert/config/log/logger.go:52     found 0 files to read from "Local FS{MatchPattern: \"/etc/alerts/*.yml\"}"
2024-07-08T04:06:08.721Z        warn    VictoriaMetrics/app/vmalert/config/log/logger.go:44     no groups found in /etc/alerts/*.yml
2024-07-08T04:11:40.949Z        warn    VictoriaMetrics/lib/httpserver/httpserver.go:433        remoteAddr: "10.10.10.10:62013"; requestURI: /vmalert/flags; unsupported path requested: "/vmalert/flags"
```

### Used command-line flags
```
--datasource.url=http://victoriametrics:8428/
--remoteRead.url=http://victoriametrics:8428/
--remoteWrite.url=http://victoriametrics:8428/
--notifier.url=http://alertmanager:9093/
--rule=/etc/alerts/*.yml
--external.alert.source=explore?orgId=1&left={"datasource":"VictoriaMetrics","queries":[{"expr{.Expr|jsonEscape|queryEscape}},"refId":"A"}],"range":{"from":"{{ .ActiveAt.UnixMilli }}","to":"now"}}
```

**Comment by jiekun** [comment]
Hi thanks for the report.

> When I navigate to the web admin page for VMAlert, the "System" urls don't work. 

I think the hyperlinks on vmalert Web UI are pointing to the right destination. They are point to the endpoint without `vmalert` prefix in both pages (`http://**:8880/vmalert/` and `http://**:8880/`).

> I think this was broken in a recent update, perhaps https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6435.

I'm not sure about it. I've tested those links (`/vmalert/metrics`) on `v1.101.0`, which is released on 2024-04-26, it seems they are not available too.

And in order to use `/vmalert/metrics` you may need command line argument like `-http.pathPrefix=vmalert`.

Feel free to attach more info here, like: In which version you are able to visit `/vmalert/metrics`?

**Comment by lukeod** [comment]
It is a strange one. I've tried v1.93.9 and v1.102.0-rc2.

It seems to be a little stranger than i thought. With v1.102.0-rc2, i can replicate the issue by:

1. Navigating to http://**:8880/
2. Clicking on "vmalert" in the upper left corner. This redirects me to ":8880/vmalert/"
3. Clicking on "Groups" in the top menu, which takes me to ":8880/vmalert/groups"
4. Clicking on "vmalert" in the upper left corner. This redirects me to ":8880/vmalert/"
5. The url for the system URLs now point at ":8880/vmalert/<x>"

This doesn't occur in the older version i tried (v1.93.9). Not sure why the page ":8880/vmalert/ " has different behavior when navigating to it from ":8880/" vs ":8880/vmalert/groups".

To be clear i don't personally mind if the url's are :8880/ or :8880/vmalert/*. For me the issue is really that the links are invalid rather than caring if "vmalert" is part of the path.

**Comment by jiekun** [comment]
Thank you. These steps can reproduce the issue. I'm investigating and see if I can fix it later.

**Comment by valyala** [comment]
FYI, this bug should be fixed in [v1.93.16 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.16). The bugfix will be also included in the latest release.

**Comment by valyala** [comment]
FYI, the bugfix for this issue has been included in [v1.97.6 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.6). The bugfix will be also included in the upcoming latest release.

**Comment by valyala** [comment]
The bugfix has been included in [v1.102.0 release](https://docs.victoriametrics.com/changelog/). Closing the issue as fixed.
